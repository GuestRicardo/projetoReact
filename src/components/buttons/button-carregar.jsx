/* eslint-disable react/react-in-jsx-scope */
import P from 'prop-types'
import { Component } from 'react';
import './style.css';

export class ButtonCarregar extends Component {
  render() {
    // eslint-disable-next-line react/prop-types
    const { text, onClick, disabled = false } = this.props;

    return (
      // eslint-disable-next-line react/react-in-jsx-scope
      <div className='buttonCenter'>
        <button className='buttonC' onClick={onClick} disabled={disabled}>{text} </button>
      </div>
    )
  }
}
ButtonCarregar.defaultProps = {
  disabled: false
};

ButtonCarregar.propTypes = {
  text: P.string.isRequired,
  onclick: P.func.isRequired,
  disabled: P.boll,
}
