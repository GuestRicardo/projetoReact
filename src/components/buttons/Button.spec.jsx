import { render, screen } from "@testing-library/react"
import { ButtonCarregar } from "./button-carregar";
import userEvent from "@testing-library/user-event";

describe('<ButtonCarregar />',()=>{
    //para garantir q o botao esta sendo carregado
    it('should render the button with the text', ()=>{
        render(<ButtonCarregar text='load more' />);

        expect.assertions(1);

        const button = screen.getByRole('button', { name: /load more/i });
        expect(button).toBeInTheDocument();
    });
    
    //para garantir q as funções sejam carregadas, click no botao    
    it('should call function on button click', ()=>{
        const fn = jest.fn();
        render(<ButtonCarregar text='load more' onClick={fn} />);

        const button = screen.getByRole('button', { name: /load more/i });
        userEvent.click(button);

        expect(fn).toHaveBeenCalledTimes(1);
    });
    //para testar o disable   
    it('should call function on button click', ()=>{
        const fn = jest.fn();
        render(<ButtonCarregar text='load more' onClick={fn} />);

        const button = screen.getByRole('button', { name: /load more/i });
        userEvent.click(button);

        expect(fn).toHaveBeenCalledTimes(1);
    });
})