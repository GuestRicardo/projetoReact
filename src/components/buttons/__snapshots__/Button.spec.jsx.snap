// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCarregar /> should match snapshot 1`] = `
<div
  class="buttonCenter"
>
  <button
    class="buttonC"
  >
    load more
  </button>
</div>
`;
